presto-session-property-managers/src/main/java/com/facebook/presto/session/SessionMatchSpec.java

```diff
@@ -78,7 +78,9 @@ public SessionMatchSpec(
             }
         }
 
-        if (resourceGroupRegex.isPresent() && !resourceGroupRegex.get().matcher(context.getResourceGroupId().toString()).matches()) {
+        if (resourceGroupRegex.isPresent() &&
+                context.getResourceGroupId().isPresent() &&
```

Comment:
It looks like it should be something like

`if (resourceGroupRegex.isPresent() && (!context.getResourceGroupId().isPresent() || !resourceGroupRegex.get().matcher(context.getResourceGroupId().get().toString()).matches()))`

Reply:
Thank you for pointing out this. Other rules inside this function use empty string when it is an empty optional object. PR is updated with the same logics.